src/gg.regression.unity.bots/Runtime/Scripts/StateRecorder/ScreenRecorder.cs

```diff
@@ -86,7 +93,12 @@ public void Awake()
             DontDestroyOnLoad(gameObject);
             _this = this;
         }
-
+        
+        public void Start()
+        {
+            _rgServiceManager = RGServiceManager.GetInstance();
+            _rgBotManager = RGBotManager.GetInstance();
+        }
```

Comment:
I think it's bad practice to get these instances and store them in static variables.. We should either access them directly each usage with `.GetInstance()` or store them in private variables.
```
private RGServiceManager _rgServiceManager;
private RGBotManager _rgBotManager;

public void Start()
{
   _rgServiceManager = RGServiceManager.GetInstance();
   _rgBotManager = RGBotManager.GetInstance();
} 
```

Reply:
If we access directly in the call sites, that will add an extra method access operator to the calls, which is why I prefer to store them in the variables.

But also, not sure if there is any functional difference for us here if we use a private static variable or a private class variable - since we are only ever creating one ScreenRecorder, I believe they are equivalent (modulo some weird C# stuff)

Reply:
That's true - probably should have marked this as a non-blocking suggestion as well